### PR TITLE
eq-381 Routing Rule Component

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

[Trello card](https://trello.com/c/WeZsp2SM/381-routing-rule-component-5)

- adds main RoutingRule component encapsulating other Routing components into a CanvasSection with toolbar, etc.
- RoutingRuleset component encapsulates one or more RoutingRules
- moved `Toolbar` and `Buttons` into a module so they can be composed elsewhere. NOTE: I would normally recommend we refactor this a bit more but there's some major design changes in the pipeline so sensible to leave it for now.

### How to review 

- check storybook examples against AC set out in the Trello card

### CODE REVIEW

File: src/components/routing/RoutingCondition.js [hunk 1]
```diff
@@ -108,7 +113,7 @@ RoutingCondition.propTypes = {
   selectedPage: CustomPropTypes.page.isRequired,
   id: PropTypes.string.isRequired,
   onPageChange: PropTypes.func.isRequired,
-  onRemoveClick: PropTypes.func.isRequired,
+  onRemoveClick: PropTypes.func,
```
Comment (WickyNilliams) [inline src/components/routing/RoutingCondition.js]: rename -> `onRemove`?

File: src/components/routing/RoutingRule.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+const RoutingRule = ({
+  children,
+  page,
+  onDeleteRule,
+  onThenChange,
+  onAddRule,
+  title,
+  sections
+}) => (
+  <React.Fragment>
```
Comment (WickyNilliams) [inline src/components/routing/RoutingRule.js]: This fragment is redundant?

File: src/components/routing/RoutingRuleEmptyMsg.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  flex-direction: column;
+  padding: 1em 0;
+`;
+
+const Title = styled.h2`
+  font-size: 1.8em;
+  font-weight: 600;
+  margin-bottom: 0.5em;
+  &::before {
+    content: url(${iconRouting});
```
Comment (WickyNilliams) [inline src/components/routing/RoutingRuleEmptyMsg.js]: do screen readers read out the url if you do this vs:

```
content: "",
background-image: `url(${iconRouting})`;
```

?

File: src/components/routing/RoutingStatement.js [hunk 4]
```diff
@@ -9,24 +9,27 @@ const CenteringColumn = styled(Column)`
...
   <Fragment>
     {children}
-    <Grid align="center">
-      <CenteringColumn gutters={false} cols={1.5}>
-        <TextButton onClick={onAddCondition} data-test="btn-add">
-          AND
-        </TextButton>
-      </CenteringColumn>
-    </Grid>
+    {onAddCondition && (
```
Comment (WickyNilliams) [inline src/components/routing/RoutingStatement.js]: When would you not pass this?

Comment (hamishtaplin) [reply src/components/routing/RoutingStatement.js]: When you can't add a condition, eg. the examples in the storybook for 'no answers' or 'unsupported answer type'.

File: src/components/routing/RoutingRuleEmptyMsg.js (hunk 3 above)
Comment (hamishtaplin) [reply src/components/routing/RoutingRuleEmptyMsg.js]: I don't think they read the url. It didn't even read a character out when we tested using it for use with inputs with a type (eg. currency)

**Pull request merged**